### PR TITLE
LPS-117507 Adds temporary fix for `hidden` element in DDM field Select

### PR DESCRIPTION
**Pull request by matuzalemsteles**
This is a quick fix so we don't have to cut a Clay CSS release now, it should be removed after we make a new Clay release https://github.com/liferay/clay/milestone/25

cc @bryceosterhaus

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
ci:test:sf

**Comment by matuzalemsteles** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/css/main.scss [hunk 1]
```diff
@@ -5,6 +5,10 @@
 @import 'image_picker';
 @import 'utilities';
 
+.ddm-field [hidden] {
+	display: none !important;
```
Comment (carloslancha) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/css/main.scss:9]: Every time @marcoscv-work sees an `!important` he cries

image

Comment (matuzalemsteles) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/css/main.scss:9]: haha sorry 😅, this is from the implementation https://github.com/liferay/clay/pull/3534

**Comment by matuzalemsteles** [comment]
ci:test:relevant

**Comment by matuzalemsteles** [comment]
Test that it was breaking, it looks ok now. https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=1243339397

- LocalFile.FormRules#ConfigureCalculateRule - OK

**Comment by matuzalemsteles** [comment]
🥺

**Comment by matuzalemsteles** [comment]
Going ahead with this!

**Comment by matuzalemsteles** [comment]
ci:forward

**Comment by wincent** [comment]
> This is a quick fix so we don't have to cut a Clay CSS release now, it should be removed after we make a new Clay release https://github.com/liferay/clay/milestone/25

Just a meta-comment here: It would be really nice if we could streamline the Clay release process to the point where it's easier to just cut a release for stuff like this (instead of applying a workaround, tracking the need to remove it, and then actually removing it afterwards; to be honest, it probably doesn't have to get streamlined _much_ to get that far). In this case, just a release of `@clayui/css`; wouldn't need to be all of Clay ([this is diff from v3.15.0 to current `HEAD`](https://gist.github.com/wincent/0feea0cd7b98d7951102519b2895542f), so the delta isn't all that big).

**Comment by matuzalemsteles** [comment]
> Just a meta-comment here: It would be really nice if we could streamline the Clay release process to the point where it's easier to just cut a release for stuff like this (instead of applying a workaround, tracking the need to remove it, and then actually removing it afterwards; to be honest, it probably doesn't have to get streamlined much to get that far). In this case, just a release of @clayui/css; wouldn't need to be all of Clay (this is diff from v3.15.0 to current HEAD, so the delta isn't all that big).

Yeah, what I talked to Bryce was that we didn't want to be fighting with CI but in the end it didn't work haha 😭

**Comment by bryceosterhaus** [comment]
@wincent ideally it would be great if it were simple to just cut a clay release and get it merged quickly, but because it takes so long to get releases merged since it runs so many tests, I figured it would be more simple to apply to fix in dxp and then update once clay is re-merged. I was going off of some [past conversation](https://github.com/liferay-frontend/liferay-portal/pull/83#issuecomment-657745864) where we seemed to think it might be better to fix small changes in dxp to alleviate failing tests and then replace later with clay.

In a perfect world, it would be great if we could make a quick npm release and get it into portal easily, but it seems to just take forever to get them merged since we have to wait on CI. If we could test CI locally quickly or get quicker feedback, we might be able to do this, but right now we are sort of dependent on running 5hrs of tests. Even a release of just one package can stem up to 200 tests on CI which takes several hours to get feedback.

**Comment by wincent** [comment]
> I was going off of some [past conversation](https://github.com/liferay-frontend/liferay-portal/pull/83#issuecomment-657745864)

Screenshot 2020-07-24 -112605-wFVdtnF3@2x

I sure am glad GitHub actually allowed you to deep-link to that comment inside the stream of 233 hidden items... 😂

I get that these things always involve trade-offs, and a lot of the levers that we might like to push to make things better aren't even ones that we can touch (CI stuff, for instance), but I think these kinds of pressure are good because they drive us — our team and other teams — to make the overall system better.

One way to look at it is that if CI is expensive, but we make smaller releases more often, we reduce the risk related to each release, and can afford to be more aggressive about manual forward (thus offsetting the cost of CI). I guess I'm trying to say that in the game of minimizing the cost of getting things merged, the rules can sometimes be bent.

It's also perfectly fine to ship workarounds like we did here if we think the overall cost is lower. I've done it myself many times. But remember that if that is our _default_ approach, then we'll never improve our release process, never make our CI better etc. That's why it's good to consider the alternatives.

**Comment by bryceosterhaus** [comment]
> But remember that if that is our default approach, then we'll never improve our release process, never make our CI better etc. That's why it's good to consider the alternatives.

Ah yes, totally agree with you. I thought you were pitching it as that we should ways just go with new releases. We actually almost did go with a new release of clay-css and merge to portal, but with the headache that other clay update PRs are causing, it seemed more risky at the time. I definitely don't think we should always do this approach, it was more of a one off instance where it seemed easier and less risky with other dependent PRs still waiting to be merged.

And I agree that these difficulties will help us push CI to be better though, I do think it's going to be tricky with clay though since we trigger so many tests when updating so many package.json files. Perhaps we can come up with a way to have more confidence in CI by being able to run a local set of tests for clay, otherwise we get stuck in these 5 hour cycles of CI failures.

cc @john-co @pablitobonito (maybe you have some ideas for testing clay update prs)

**Comment by john-co** [comment]
For this PR, the first `ci:test:relevant` had no unique failures so there wasn't a need to retest and could be easily `ci:forward`. For the number of tests here...I think we'd have to check with Forms/DDL team to reevaluate. But since we're touching a main.css file...that could potentially affect all those tests and our `ci:test:relevant` filters isn't smart enough to tell which specific tests to run.

If there are pulls where we are just updating package.json files, I was already thinking of creating a much smaller test suite for clay updates that could replace `ci:test:relevant` process: https://issues.liferay.com/browse/LRQA-60383. Are there such cases that we only update package.json files or does it always involve modifying the src files of the submodules?

**Pull request closed without merging**